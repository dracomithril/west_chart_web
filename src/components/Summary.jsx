/**
 * Created by Gryzli on 09.04.2017.
 */
import React from 'react';
import PropTypes from 'prop-types';
import copy from 'clipboard-copy';
import { Badge, Button } from 'react-bootstrap';
import './components.css';
// let {sorting} = require('./../utils');
const create_print_list = (elem, index) => (
  <div key={elem.id}>
    <span>{index + 1}</span>
    {`. ${elem.link.title} `}
    <Badge bsClass="likes">{`${elem.reactions_num} likes`}</Badge>
  </div>
);
// const create_print_list_txt = (elem, index) => {
//     return `${index + 1}. ${elem.link.title} ${elem.reactions_num} likes`
// };
export default class Summary extends React.Component {
  state = {
    introText: '',
    riddleText: '',
    riddleUrl: '',
  };

  /* istanbul ignore next */
  componentDidMount() {
    console.info('component Summary did mount');
  }

  /* istanbul ignore next */
  componentWillUnmount() {
    console.info('component Summary unmounted');
  }

  onCopyToClipboard = () => {
    const { store } = this.context;
    const { sp_playlist_info } = store.getState();
    const { selected } = this.props;
    const playList = selected
      .map((elem, ind) => `${ind + 1}. ${elem.link.title} ${elem.reactions_num} likes`)
      .join('\n');
    const text = `[WCS Weekly Westletter]

${this.state.introText}

${playList}

${sp_playlist_info.url ? `Link to spotify playlist:${sp_playlist_info.url}` : 'No link'}

Riddle:
${this.state.riddleText}

${this.state.riddleUrl}`;
    copy(text);
    alert('Summary was copied to clipboard');
  };

  render() {
    const { store } = this.context;
    const { sp_playlist_info } = store.getState();
    const print_list = this.props.selected.map(create_print_list);
    return (
      <div className="summary">
        <h3 id="summary">
          Summary
          <Button bsStyle="info" onClick={this.onCopyToClipboard}>
            <i className="fa fa-clipboard" />
          </Button>
          <Button
            onClick={() => {
              alert('Not implemented jet.');
            }}
            disabled
          >
            Publish2<i className="fa fa-facebook-official" />
          </Button>
        </h3>
        <h6>[WCS Weekly Westletter]</h6>
        <textarea
          id="textarea_add"
          className="write_your_mind"
          placeholder="Here write what you want"
          defaultValue={this.state.introText}
          onChange={e => {
            this.setState({ introText: e.target.value });
          }}
        />
        {print_list.length === 0 && (
          <div>
            <span style={{ color: 'red' }}>Here will be list of tracks your choosing</span>
          </div>
        )}
        <div id="popover-contained" title="Print list">
          {print_list}
        </div>

        {/* <textarea id="popover-contained" title="Print list" className="write_your_mind " placeholder={"Print list"} value={print_list.join('\n')}/> */}
        <h6>
          {'Link to spotify playlist: '}
          {sp_playlist_info.url && (
            <a href={sp_playlist_info.url} target="_newtab">
              {sp_playlist_info.url}
            </a>
          )}
          {!sp_playlist_info.url && <span style={{ color: 'red' }}>No link</span>}
        </h6>
        <textarea
          id="riddler"
          className="write_your_mind"
          placeholder="riddle that you have in mind"
          defaultValue={this.state.riddleText}
          onChange={({ target }) => {
            this.setState({ riddleText: target.value });
          }}
        />
        <br />
        <label htmlFor="link2riddle">
          Link for Riddle
          <input
            type="text"
            id="link2riddle"
            placeholder=" link to riddle"
            value={this.state.riddleUrl}
            onChange={e => {
              this.setState({ riddleUrl: e.target.value });
            }}
          />
        </label>
      </div>
    );
  }
}
Summary.contextTypes = {
  store: PropTypes.object,
};
Summary.propTypes = {
  selected: PropTypes.array,
};
