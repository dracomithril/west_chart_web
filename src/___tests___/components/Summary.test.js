/**
 * Created by XKTR67 on 4/19/2017.
 */
import React from 'react';
import PropTypes from 'prop-types';
import Summary from './../../components/Summary';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {shallowToJson} from 'enzyme-to-json';
import configureMockStore from 'redux-mock-store';

Enzyme.configure({ adapter: new Adapter() });
const initial_state = require('../data/initial_state.json');

const mockStore = configureMockStore([]);


describe('<Summary/>', () => {
  it('renders without crashing', () => {
    const store = mockStore(initial_state);
    const wrapper = shallow(
      <Summary selected={[]}/>, {
        context: { store },
        childContextTypes: { store: PropTypes.object }
      }
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();

  });
});
