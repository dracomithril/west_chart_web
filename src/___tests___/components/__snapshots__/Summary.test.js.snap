// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Summary/> renders without crashing 1`] = `
<div
  className="summary"
>
  <h3
    id="summary"
  >
    Summary
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="info"
      disabled={false}
      onClick={[Function]}
    >
      <i
        className="fa fa-clipboard"
      />
    </Button>
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      disabled={true}
      onClick={[Function]}
    >
      Publish2
      <i
        className="fa fa-facebook-official"
      />
    </Button>
  </h3>
  <h6>
    [WCS Weekly Westletter]
  </h6>
  <textarea
    className="write_your_mind"
    defaultValue=""
    id="textarea_add"
    onChange={[Function]}
    placeholder="Here write what you want"
  />
  <div>
    <span
      style={
        Object {
          "color": "red",
        }
      }
    >
      Here will be list of tracks your choosing
    </span>
  </div>
  <div
    id="popover-contained"
    title="Print list"
  />
  <h6>
    Link to spotify playlist: 
    <span
      style={
        Object {
          "color": "red",
        }
      }
    >
      No link
    </span>
  </h6>
  <textarea
    className="write_your_mind"
    defaultValue=""
    id="riddler"
    onChange={[Function]}
    placeholder="riddle that you have in mind"
  />
  <br />
  <label
    htmlFor="link2riddle"
  >
    Link for Riddle
    <input
      id="link2riddle"
      onChange={[Function]}
      placeholder=" link to riddle"
      type="text"
      value=""
    />
  </label>
</div>
`;
